akka-projection-testkit/src/main/scala/akka/projection/testkit/javadsl/ProjectionTestKit.scala

```diff
@@ -90,11 +88,9 @@ final class ProjectionTestKit private[akka] (testKit: ActorTestKit) {
 
     val probe = testKit.createTestProbe[Nothing]("internal-projection-testkit-probe")
 
-    val settingsForTest = ProjectionSettings(system).copy(RestartBackoffSettings(0.millis, 0.millis, 0.0, 0))
-
     val running =
       projection
-        .withSettings(settingsForTest)
+        .withRestartBackoff(0.millis, 0.millis, 0.0, 0)
```

Comment:
Note that this will override any RestartBackoffSettings that the user has defined at the Projection level, while the previous was only overriding the default settings (instead of config). Maybe it never makes sense to run with any other settings than this, and then all good.

Reply:
I don't get your point. 

Previously we were building a new settings and overriding with a new backoff settings, the one that make sense for that specific way of running the tests. 

The new code is only overriding the backoff part. Actually, is the other way around.

Reply:
Previously it was possible for the user to override the backoff settings in the test and actually have restart backoff. Maybe that never makes sense, so all good.

Reply:
I don't think it was possible, we were loading the config and then overriding it with what we needed to make it work in test. 

```scala
ProjectionSettings(system).copy(RestartBackoffSettings(0.millis, 0.millis, 0.0, 0))	
```

Any user settings added programatically was overriden and any config for backoff was also overidden. 

Anyway, I will add a notice in the scaladoc and documentation about it. Users must understand that backoff can't be enabled for tests.